core/src/main/scala/kafka/server/metadata/DynamicClientQuotaPublisher.scala

```diff
@@ -48,6 +51,11 @@ class DynamicClientQuotaPublisher(
   ): Unit = {
     val deltaName = s"MetadataDelta up to ${newImage.highestOffsetAndEpoch().offset}"
     try {
+        val clientQuotaCallback = conf.getConfiguredInstance(QuotaConfig.CLIENT_QUOTA_CALLBACK_CLASS_CONFIG, classOf[ClientQuotaCallback])
+        if (clientQuotaCallback != null) {
```

Comment:
Should we also check `clientQuotaCallback instance DefaultQuotaCallback`?

Reply:
I don't think we need to check for `DefaultQuotaCallback`, because the default value of `QuotaConfig.CLIENT_QUOTA_CALLBACK_CLASS_CONFIG` is `null`, not `DefaultQuotaCallback`.

Reply:
https://github.com/apache/kafka/blob/6836fa259e0c50cc6a8d5099b3fbbacda9c2aadc/core/src/main/scala/kafka/server/ClientQuotaManager.scala#L151

Here will initialize to `DefaultQuotaCallback ` if I am correct.

Reply:
Hmm, user should set this config in the server, it will have the not null value, user also can extend the `DefaultQuotaCallback` and override the method, make sense?
https://github.com/apache/kafka/blob/5a4590c02016faeea3adeadb0d53e4388a558096/server-common/src/main/java/org/apache/kafka/server/config/QuotaConfig.java#L117

Reply:
Yes, you are right. However, what I pointed out is the closest path to create `quotaCallback` so even if it is set null, `DefaultQuotaCallback` will be created.

> it will have the not null value, user also can extend the DefaultQuotaCallback and override the method, make sense?

Make sense. Maybe we can use `quotaCallback .getClass() == DefaultQuotaCallback.class` to avoid this problem?

Reply:
This `DefaultQuotaCallback` is `ClientQuotaManager` private inner class, I still don't want to expose this class.

Reply:
And [client.quota.callback.class](https://kafka.apache.org/documentation/#brokerconfigs_client.quota.callback.class) this config can setting `kafka.server.ClientQuotaManager#DefaultQuotaCallback`? 
I think DefaultQuotaCallback the main function is setting the default value, not execute the callback.